addons/point_of_sale/models/pos_order.py

```diff
@@ -1101,6 +1292,11 @@ def _prepare_refund_data(self, refund_order_id):
 
     @api.model
     def create(self, values):
+        if values.get('server_id'):
```

Comment:
don't do this override in create, the check is of a value 'server_id' must be done before the call to create. Call a write or a create depending on if there is  a server_id in the order_ui dictionnary. The value 'server_id' should never be passed to the create or write, as the field doesn't exists on the model.